controllers/vspherevm_controller.go

```diff
@@ -473,7 +473,7 @@ func (r vmReconciler) reconcileNormal(ctx context.Context, vmCtx *capvcontext.VM
 func (r vmReconciler) isWaitingForStaticIPAllocation(vmCtx *capvcontext.VMContext) bool {
 	devices := vmCtx.VSphereVM.Spec.Network.Devices
 	for _, dev := range devices {
-		if !dev.DHCP4 && !dev.DHCP6 && len(dev.IPAddrs) == 0 && len(dev.AddressesFromPools) == 0 {
+		if !dev.DHCP4 && !dev.DHCP6 && len(dev.IPAddrs) == 0 && len(dev.AddressesFromPools) == 0 && !dev.IPConfigurationNotMandatory {
```

Comment:
Can we have a comment here explaining this long if? Its hard to read and remember again.

We should also adjust the godoc of the function.

Maybe there's a way to write this function in a way so its more readable: (please check if the logic is correct)

```
// TODO explain
if dev.IgnoreIPAllocation {
  continue
}

// TODO explain
if len(dev.IPAddrs) > 0 || len(dev.AddressesFromPools) > 0 {
  continue
}

// TODO explain
if dev.DHCP4 || dev.DHCP6 {
  continue
}

// TODO explain
return true
```

Reply:
I've split this if statement into 3 separate statements as by tour suggestion.